docs/milestone2.md

```diff
@@ -103,40 +101,23 @@ lambda x, y: pyad.cos(x) * pyad.sin(y)
 
 **pyad** should be able to deal with either of these cases and end up with the same result, hence allowing the user to build functions of arbitrary complexity and not worry about having to change the implementation method.
 
-The core operating principle will be:
-
-1. Define a function to be differentiated (or alternatively use a lambda function inline) `def: test_function(x) return 2 * x` or `lambda x: 2 * x`
-2. Instantiate a specific class of Automatic Differentiation from the **pyad** package (for instance `my_ad = pyad.forwardmode(test_function)` or `my_ad = pyad.forwardmode(lambda x: 2 * x)` 
-3. Set up parameters with which to differentiate - this will need to be the seed of each of the variables and the initial derivative conditions (usually 1). `my_ad.initial_conditions(2.5), my_ad.derivative_seeds(1)` or `my_ad.initial_conditions({x:2.5, y:5.8}), my_ad.derivative_seeds({x:1, y:1)`
-4. Compute the derivative - this should be an explicit step as the computational time may be non-negligible. `my_ad.compute_derivative()` - this computation could have a number of options
-    * Allow the user control of whether or not to keep the full trace table for inspection (a must for reverse mode).
-    * Utilise dual number implementation.
-5. There should be a number of methods for interrogating the result once this is completed, these could include:
-    * Get the output value for the derivative
-    * View the initial conditions
-    * View the output trace table
-
-
-### Example Use Case
-
-The following is an example of how to use the pyad package to differentiate a user defined function using the forward mode. The specific code is yet to be implemented but the operating process will be as follows:
+### Demo
 
 ```python
 import pyad
 
-def simple_function(x, y):
-    return 2 * pyad.sin(x) + pyad.cos(y + 4)
+x = pyad.Variable('x', 1)
+y = pyad.Variable('y', 2)
+z = pyad.Variable('z', 3)
 
-ad_forward.pyad.forwardmode(simple_function)
-ad_forward.initial_conditions({x:3,y:0.5})
-ad_forward.derivative_seeds({x:1,y:1})
+>>> x**2 + 2*y + z
+Tensor(8, D(z=1, x=2, y=2))
 
-simple_derivative = ad_forward.compute_derivative(trace=True)
+def test_fun(x, y, z):
+	return pyad.exp(pyad.cos(x) + pyad.sin(y))**z
 
-# Options to get information out of the object after computation
-simple_derivative = ad.foward.get_derivative()
-trace_df = ad.foward.get_trace()
-initial_conditions = ad.foward.get_initial_conditions()
+>>> test_fun(x,y,z):
+Tensor(77.4, D(x=-195, y=-96.6, z=112))
```

Comment:
Maybe something like this for this example to show how to extract specific derivatives with the `d` attribute:
```
>>> res = test_fun(x,y,z)
>>> res.d['x']
-195
>>> res.d['y']
-96.6
>>> res.d['z']
112
```